sopel/modules/help.py

```diff
@@ -12,10 +12,81 @@
 
 import collections
 import requests
+import socket
 import textwrap
 
 from sopel.logger import get_logger
 from sopel.module import commands, rule, example, priority
+from sopel.config.types import (
+    StaticSection, ChoiceAttribute
+)
+
+
+# Pastebin handlers
+
+
+def post_to_clbin(bot, msg):
+    result = requests.post('https://clbin.com/', data={'clbin': msg})
+    result = result.text
+
+    if 'https://clbin.com/' in result:
+        return result
+    else:
+        LOGGER.error("Invalid result %s", result)
+        raise PostingException()
+
+
+def post_to_0x0(bot, msg):
+    payload = {'file': msg}
+    result = requests.post('https://0x0.st', files=payload)
+    return result.text
+
+
+def post_to_hastebin(bot, msg):
+    result = requests.post('https://hastebin.com/documents', data=msg)
+    result = result.json()
+    if 'key' not in result:
+        LOGGER.error("Invalid result %s", result)
+        raise PostingException()
+    return "https://hastebin.com/" + result['key']
+
+
+def post_to_termbin(bot, msg):
+    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
```

Comment:
Because `socket` is so low-level, theres a lot more things that need to be configured here to be presentable to the user; specifcally, a timeout [`s.settimeout`](https://docs.python.org/2/library/socket.html#socket.socket.settimeout) needs to be set so that the bot doesn't try to wait for a connection forever
```suggestion
    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
    s.settimeout(10)  # raise `socket.timeout` if socket doesn't respond for >10 seconds
```

Don't forget to handle this exception nicely (see previous comment).